controllers/discovery/initDiscoveryDomainData.js

```diff
@@ -45,11 +56,21 @@ const InitDiscoveryDomainData = async (req, res) => {
     });
     let suggestedDomains = domainWithCommonFollowerResult;
 
+    let totalData = await sequelize.query(totalDataQuery, {
+      type: QueryTypes.SELECT,
+      replacements: {
+        userId
+      },
+      raw: true
+    });
+    totalData = totalData[0]?.total_data;
+
     return res.status(200).json({
       success: true,
       message: `Fetch discovery data success`,
       suggestedDomains,
-      page: page + 1
+      page: page + 1,
+      total_page: Math.ceil(totalData / limit)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling is missing in this section. If an error occurs during the execution of the SQL query, it will not be handled properly and may cause the application to crash. Consider adding a try-catch block around the SQL query execution and handle any potential errors appropriately. Also, consider checking if `totalData` is not undefined before calculating `total_page` to avoid potential division by zero errors.
```diff
- totalData = totalData[0]?.total_data;
+ totalData = totalData && totalData[0] ? totalData[0].total_data : 0;

- return res.status(200).json({
-   success: true,
-   message: `Fetch discovery data success`,
-   suggestedDomains,
-   page: page + 1,
-   total_page: Math.ceil(totalData / limit)
- });
+ try {
+   return res.status(200).json({
+     success: true,
+     message: `Fetch discovery data success`,
+     suggestedDomains,
+     page: page + 1,
+     total_page: totalData > 0 ? Math.ceil(totalData / limit) : 0
+   });
+ } catch (error) {
+   return res.status(500).json({
+     success: false,
+     message: 'An error occurred while fetching discovery data',
+     error: error.message
+   });
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->